### PR TITLE
[Core] add QKV fusion to AuraFlow and PixArt Sigma

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

As a follow-up of https://github.com/huggingface/diffusers/pull/8829, I decided to give QKV fusion a try for AuraFlow and PixArt-Sigma when doing int8 quantization. Some nice gains on an H100:

### PixArt

* Without fusion: 1.499 seconds
* With fusion: 1.455 seconds

### AuraFlow

* Without fusion: 13.850 seconds
* With fusion: 13.707 seconds

### CODE REVIEW

File: src/diffusers/models/attention_processor.py [hunk 1]
```diff
@@ -219,6 +219,7 @@ def __init__(
             self.to_v = None
 
         if self.added_kv_proj_dim is not None:
+            self.added_proj_bias = added_proj_bias
```
Comment (yiyixuxu) [inline src/diffusers/models/attention_processor.py:222]: ok but I think it's easier if you move this outside of the if, along with other attributes - that way `Attention` will always has this attribute

File: src/diffusers/models/attention_processor.py [hunk 2]
```diff
@@ -1261,6 +1265,103 @@ def __call__(
...
+                "FusedAuraFlowAttnProcessor2_0 requires PyTorch 2.0, to use it, please upgrade PyTorch to at least 2.1 or above as we use `scale` in `F.scaled_dot_product_attention()`. "
+            )
+
+    def __call__(
+        self,
+        attn: Attention,
+        hidden_states: torch.FloatTensor,
+        encoder_hidden_states: torch.FloatTensor = None,
+        *args,
+        **kwargs,
```
Comment (yiyixuxu) [inline src/diffusers/models/attention_processor.py:1283]: we do not need them, no?
```suggestion
```

**Review by yiyixuxu — APPROVED** [review]
very nice! thank you

**Comment by sayakpaul** [comment]
@yiyixuxu because we have

https://github.com/huggingface/diffusers/blob/50e66f2f9560dec6a6e0fa2d50cdb14f1ae56232/src/diffusers/models/attention_processor.py#L480

I had to do:
https://github.com/huggingface/diffusers/pull/8952/commits/33310efcdfbf24d8c168834811d78359ede79681

LMK if you have other ideas to counter it. Without it, we will have:
https://github.com/huggingface/diffusers/actions/runs/10104445763/job/27943457611?pr=8952#step:7:17774

**Comment by sayakpaul** [comment]
@yiyixuxu a friendly ping.

**Comment by yiyixuxu** [comment]
@sayakpaul you're right thank you!

**Pull request merged**